internal/search/query/parser.go

```diff
@@ -122,8 +122,15 @@ func skipSpace(buf []byte) int {
 }
 
 type heuristic struct {
-	parensAsPatterns    bool // if true, parses parens as patterns rather than expression groups.
-	allowDanglingParens bool // if true, disables parsing parentheses as expression groups.
+	// if true, parses parens as patterns rather than expression groups.
+	parensAsPatterns bool
+	// if true, disables parsing parentheses as expression groups.
+	allowDanglingParens bool
+	// if true, the parser accepts any characters (e.g., unbalanced quotes,
```

Comment:
nit: since touched these lines, good chance to fix the style
```suggestion
	// If true, parses parens as patterns rather than expression groups.
	parensAsPatterns bool
	// If true, disables parsing parentheses as expression groups.
	allowDanglingParens bool
	// If true, the parser accepts any characters (e.g., unbalanced quotes,
```